### PR TITLE
Soil map extraction

### PR DESCRIPTION
**Pull request by hansvancalster**
This PR adds the function extract_soil_map_data().

@niconoe @damianooldoni After adding this function and running checks on the package, I noticed :
image

Oddly, but when I run just the script containing the test, all tests pass. Don't know what is going on really....

### CODE REVIEW

**Comment by hansvancalster** [comment]
@florisvdh travis build failed. Do you know anything about Travis?

**Comment by damianooldoni** [comment]
Thanks @hansvancalster , strange. I will check it.
About Travis, error seems in installing package `pkgdown`. But failure in tests will cause a Travis failure as well.

**Comment by florisvdh** [comment]
I can have a look, sounds familiar

**Comment by florisvdh** [comment]
@damianooldoni the build is still on xenial. I'll make a fix (a PR to this branch) to migrate it to Bionic.

**Comment by damianooldoni** [comment]
I don't see problems while running `devtools:test()` after switching to `soil-map-extraction` branch. Strange.

test_result_inborutils

**Comment by florisvdh** [comment]
@damianooldoni the `pkgdown` problem had to do with missing libraries on the build machine, which still ran Ubuntu Xenial. Let's wait whether PR #95 build (on Ubuntu Bionic) succeeds, and when successful, merge into `soil-map-extraction`. Running now: https://travis-ci.com/github/inbo/inborutils/builds/214125615

**Comment by hansvancalster** [comment]
> I don't see problems while running `devtools:test()` after switching to `soil-map-extraction` branch. Strange.
> 
> test_result_inborutils

I see the same when running `devtools::test()` but not when I run devtools::check()

**Comment by florisvdh** [comment]
This gives some hope - this is the result here - and it seems like an interesting note:

```
── R CMD check results ────────────────────────────── inborutils 0.1.0.9086 ────
Duration: 2m 33.3s

> checking package dependencies ... NOTE
  Imports includes 26 non-default packages.
  Importing from so many packages makes the package vulnerable to any of
  them becoming unavailable.  Move as many as possible to Suggests and
  use conditionally.

0 errors ✓ | 0 warnings ✓ | 1 note x

R CMD check succeeded

```

**Comment by florisvdh** [comment]
Solved in PR #95

File: R/extract_soil_map_data.R [hunk 1]
```diff
@@ -0,0 +1,62 @@
+#' @title Extract soil properties from Flemish soil map
+#'
+#' @description This function queries the Flemish soil map attributres at a given coordinate,
+#' http://www.geopunt.be/catalogus/datasetfolder/5c129f2d-4498-4bc3-8860-01cb2d513f8f
```
Comment (florisvdh) [inline R/extract_soil_map_data.R]: I don't think this URL (and the one on line 7) will be clickable in this way. Check https://roxygen2.r-lib.org/articles/formatting.html#links

File: R/extract_soil_map_data.R [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+#' by using the affiliated WFS service provided by DOV. The user can pick the
+#' properties of interest. A full list of properties is available at
+#' https://www.dov.vlaanderen.be/geoserver/bodemkaart/bodemtypes/wfs?request=DescribeFeatureType
+#' Coordinates should be given as Lambert 72 (epsg 31370). When outside the
```
Comment (florisvdh) [inline R/extract_soil_map_data.R]: See https://epsg.org/search/by-name?sessionkey=lqlp8f7hg1&searchedTerms=lambert+72 : the name is best put more specifically.

```suggestion
#' Coordinates should be given in the 'Belge 1972 / Belgian Lambert 72' 
#' (\href{https://epsg.io/31370}{EPSG:31370}) coordinate reference
#' system. When outside the
```

File: R/extract_soil_map_data.R [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+#' Flemish region, an NA value is given for each of the properties.
+#'
+#' @param x_lam The numeric value of the X coordinate as Lambert 72.
+#' @param y_lam The numeric value of the X coordinate as Lambert 72.
```
Comment (florisvdh) [inline R/extract_soil_map_data.R]:
```suggestion
#' @param x_lam The numeric value of the X coordinate in CRS 
#' 'Belge 1972 / Belgian Lambert 72' (EPSG:31370).
#' @param y_lam The numeric value of the Y coordinate in CRS 
#' 'Belge 1972 / Belgian Lambert 72' (EPSG:31370).
```

**Comment by hansvancalster** [comment]
`devtools::check()` still gives me the same error.  So likely this is some Windows thing or a specific issue on my laptop since it seems to run fine in Travis CI?

File: R/extract_soil_map_data.R [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+               propertyname = as.character(paste(properties_of_interest,
+                                                 collapse = ",")),
+               CRS = "EPSG:31370",
+               CQL_FILTER = sprintf("INTERSECTS(geom,POINT(%s %s))",
+                                    x_lam, y_lam)) # INTERSECT OPERATOR
+
+  result <- GET(wfs_bodemtypes, query = query)
+  parsed <- fromJSON(content(result, "text"))
+  soil_info_df <- parsed$features
+  # if else toegevoegd om gevallen op te vangen waarbij een punt (net) buiten de kaart valt
```
Comment (florisvdh) [inline R/extract_soil_map_data.R]: replace with English comment?

**Review by florisvdh — APPROVED** [review]
@hansvancalster I tested the function from the tutorial yesterday, so I guess it still works :wink: 

Some minor documentation suggestions; please build Rd files again.

**Pull request merged**